core/kazoo_provisioner/src/provisioner_v5.erl

```diff
@@ -218,27 +220,51 @@ set_owner(Device) ->
     end.
 
 %%------------------------------------------------------------------------------
-%% @doc
+%% @doc Calculate the settings for the device based on the merge down
+%%
+%% When configuring a device, settings from the device, user, and account must be
+%% merged down into a unified view of what is sent to the provisioner.
+%%
+%% For instance, when a user is disabled, the device(s) will be disabled, but if
+%% the device is re-saved (and still owned by the disabled user) and enabled, it
+%% needs to continue to be disabled on provisioner.
+%%
+%% This function should create the merged version of the device's settings in a
+%% way similar to kz_endpoint when building bridge strings.
 %% @end
 %%------------------------------------------------------------------------------
--spec settings(kzd_devices:doc()) -> kz_json:object().
-settings(Device) ->
-    settings(Device, kz_doc:type(Device)).
-
--spec settings(kzd_devices:doc(), kz_term:ne_binary()) -> kz_json:object().
-settings(Device, <<"device">>) ->
-    Props = props:filter_empty([{<<"lines">>, settings_lines(Device)}
-                               ,{<<"codecs">>, settings_codecs(Device)}
-                               ,{<<"datetime">>, settings_datetime(Device)}
-                               ,{<<"feature_keys">>, settings_feature_keys(Device)}
-                               ,{<<"line_keys">>, settings_line_keys(Device)}
-                               ,{<<"combo_keys">>, settings_combo_keys(Device)}
-                               ]),
-    kz_json:from_list(Props);
-settings(Device, _) ->
-    Props = props:filter_empty([{<<"lines">>, settings_lines(Device)}
-                               ]),
-    kz_json:from_list(Props).
+-spec settings(doc()) -> kz_json:object().
+settings(Doc) ->
```

Comment:
ok, so settings/1 should be called within the context of a device doc which should get the user/account settings and do its thing and if a user, should fetch owned_by devices and loop over them and if the property is already set at the account level in provisioner then don't send it on the device update.